src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -84,6 +85,10 @@ protected void runTool( JavaSparkContext ctx ) {
             throw new UserException.Require2BitReferenceForBroadcast();
         }
 
+        if (joinStrategy == JoinStrategy.OVERLAPS_PARTITIONER && getHeaderForReads().getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new UserException.BadInput("Reads must be coordinate sorted when using the overlaps partitioner join strategy.");
+        }
```

Comment:
Do we need an equivalent check in `BQSRPipelineSpark`? Should this check be pushed down into `AddContextDataToReadSpark` so that you don't have to duplicate it in both places?